[jdcbc-v2] Connection to IPv6 URL fails Regex

<!--
    possible labels: bug,client-api-v2,jdbc-v2
-->
## Description
Attempting to pass a JDBC URL with an IPv6 address in 0.8.x clients fails due to the validation regex.


### Steps to reproduce
1. Create a new data source like `new ClickHouseDataSource("jdbc:clickhouse://[::1]:8123", properties);`
2. Attempt a JDBC operation.

### Error Log or Exception StackTrace

```
Caused by: java.sql.SQLException: Invalid url //[::1]:8123

	at com.clickhouse.jdbc.internal.JdbcConfiguration.parseUrl(JdbcConfiguration.java:144)
	at com.clickhouse.jdbc.internal.JdbcConfiguration.<init>(JdbcConfiguration.java:59)
```

### Expected Behaviour
The URL should be accepted and allow connection.

### Code Example

```java
    String url = "jdbc:clickhouse://[::1]:8123";
    Properties properties = new Properties();
    properties.put("user", "default");
    properties.put("password", "");
    properties.put("database", "default");
    ClickHouseDataSource dataSource = new ClickHouseDataSource(url, properties);
    ResultSet catalogs = dataSource.getConnection().getMetaData().getCatalogs();
```

### Configuration

#### Client Configuration

See code example.

#### Environment
* on-prem
* Client version: 0.8.6
* Language version: Java 21
* OS: Mac OS

#### ClickHouse Server
* ClickHouse Server version: 25.3.2.39
* ClickHouse Server non-default settings, if any:

--- a/jdbc-v2/src/main/java/com/clickhouse/jdbc/internal/JdbcConfiguration.java
+++ b/jdbc-v2/src/main/java/com/clickhouse/jdbc/internal/JdbcConfiguration.java
@@ -126,7 +126,7 @@ public class JdbcConfiguration {
      *     <li>5 - query parameters as is (optional)</li>
      * </ul>
      */
-    private static final Pattern URL_REGEXP = Pattern.compile("(https?:)?\\/\\/([\\w\\.\\-]+):?([\\d]*)(?:\\/([\\w]+))?\\/?\\??(.*)$");
+    private static final Pattern URL_REGEXP = Pattern.compile("(https?:)?\\/\\/([\\w\\.\\-]+|\\[[0-9a-fA-F:]+\\]):?([\\d]*)(?:\\/([\\w]+))?\\/?\\??(.*)$");
 
     /**
      * Extracts positions of parameters names.

--- a/jdbc-v2/src/test/java/com/clickhouse/jdbc/internal/JdbcConfigurationTest.java
+++ b/jdbc-v2/src/test/java/com/clickhouse/jdbc/internal/JdbcConfigurationTest.java
@@ -40,6 +40,9 @@ public class JdbcConfigurationTest {
 
         return new Object[][] {
                 {"jdbc:clickhouse://localhost:8123/", "http://localhost:8123", defaultProps, defaultParams},
+                {"jdbc:clickhouse://127.0.0.1:8123/", "http://127.0.0.1:8123", defaultProps, defaultParams},
+                {"jdbc:clickhouse://[::1]:8123/", "http://[::1]:8123", defaultProps, defaultParams},
+                {"jdbc:clickhouse://[::1]/", "http://[::1]", defaultProps, defaultParams},
                 {"jdbc:clickhouse://localhost:8443/clickhouse?param1=value1&param2=value2", "http://localhost:8443", defaultProps, simpleParams},
                 {"jdbc:clickhouse:https://localhost:8123/clickhouse?param1=value1&param2=value2", "https://localhost:8123", defaultProps, simpleParams},
                 {"jdbc:clickhouse://localhost:8443/", "https://localhost:8443", useSSL, useSSLParams},

